### PR TITLE
Add post job to build canary build on latest master

### PR DESCRIPTION
**Pull request by cofyc**
It also adds unit test to restrict usage of secrets.

xref: https://github.com/kubernetes-sigs/sig-storage-local-static-provisioner/issues/9

### CODE REVIEW

**Comment by cofyc** [comment]
/assign @fejta @BenTheElder 
cc @msau42

File: config/jobs/kubernetes-sigs/sig-storage-local-static-provisioner/sig-storage-local-static-provisioner-trusted.yaml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+      containers:
+      - image: gcr.io/k8s-testimages/kubekins-e2e:v20190108-eadd44c98-master
+        command:
+        - runner.sh
+        args:
+        - make
+        - e2e
+        - release
+        env:
+        - name: PROVIDER
```
Comment (msau42) [inline config/jobs/kubernetes-sigs/sig-storage-local-static-provisioner/sig-storage-local-static-provisioner-trusted.yaml]: don't need these 2 envs?

File: config/jobs/kubernetes-sigs/sig-storage-local-static-provisioner/sig-storage-local-static-provisioner-trusted.yaml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        # docker-in-docker needs privileged mode
+        securityContext:
+          privileged: true
+        volumeMounts:
+        - name: pusher-docker-config
+          mountPath: /etc/pusher-docker-config
+      volumes:
+      - name: pusher-docker-config
+        secret:
+          secretName: sig-storage-local-static-provisioner-pusher
```
Comment (fejta) [inline config/jobs/kubernetes-sigs/sig-storage-local-static-provisioner/sig-storage-local-static-provisioner-trusted.yaml]: I think we still need to create this? Contact me to set it up.

File: config/jobs/kubernetes-sigs/sig-storage-local-static-provisioner/sig-storage-local-static-provisioner-trusted.yaml (hunk 1 above)
Comment (cofyc) [reply config/jobs/kubernetes-sigs/sig-storage-local-static-provisioner/sig-storage-local-static-provisioner-trusted.yaml]: You mean why I add these 2 envs? It runs e2e before building image, I want to make it clearer.

Comment (msau42) [reply config/jobs/kubernetes-sigs/sig-storage-local-static-provisioner/sig-storage-local-static-provisioner-trusted.yaml]: thanks for the clairifcation, this lgtm
/approve

File: config/jobs/kubernetes-sigs/sig-storage-local-static-provisioner/sig-storage-local-static-provisioner-trusted.yaml (hunk 2 above)
Comment (cofyc) [reply config/jobs/kubernetes-sigs/sig-storage-local-static-provisioner/sig-storage-local-static-provisioner-trusted.yaml]: I've sent json file to your email. You can check email from cofyc.jackson@gmail.com.

File: config/jobs/kubernetes-sigs/sig-storage-local-static-provisioner/sig-storage-local-static-provisioner-trusted.yaml (hunk 1 above)
Comment (cofyc) [reply config/jobs/kubernetes-sigs/sig-storage-local-static-provisioner/sig-storage-local-static-provisioner-trusted.yaml]: e2e target is removed

**Pull request merged**